flink-connectors/flink-hbase/src/main/java/org/apache/flink/table/descriptors/HBase.java

```diff
@@ -0,0 +1,118 @@
+import static org.apache.flink.table.descriptors.HBaseValidator.CONNECTOR_ZK_NODE_PARENT;
+import static org.apache.flink.table.descriptors.HBaseValidator.CONNECTOR_ZK_QUORUM;
+
+/**
+ * Connector descriptor for Apache HBase.
+ */
+public class HBase extends ConnectorDescriptor {
+	private DescriptorProperties properties = new DescriptorProperties();
+
+	public HBase() {
+		super(CONNECTOR_TYPE_VALUE_HBASE, 1, true);
+	}
+
+	/**
+	 * Set the Apache HBase version to be used. Optional.
+	 *
+	 * @param version HBase version. E.g., "1.4.3".
+	 */
+	public HBase version(String version) {
+		properties.putString(CONNECTOR_VERSION, version);
+		return this;
+	}
+
+	/**
+	 * Set the HBase table name, Required.
+	 *
+	 * @param tableName Name of HBase table.  E.g., "testNamespace:testTable", "testDefaultTable"
+	 */
+	public HBase tableName(String tableName) {
+		properties.putString(CONNECTOR_TABLE_NAME, tableName);
+		return this;
+	}
+
+	/**
+	 * Set the zookeeper quorum address to connect the HBase cluster. Required.
+	 *
+	 * @param zookeeperQuorum zookeeper quorum address to connect the HBase cluster. E.g., "localhost:2181,localhost:2182,localhost:2183".
+	 */
+	public HBase zookeeperQuorum(String zookeeperQuorum) {
+		properties.putString(CONNECTOR_ZK_QUORUM, zookeeperQuorum);
+		return this;
+	}
+
+	/**
+	 * Set the zookeeper node parent path of HBase cluster. Optional.
+	 *
+	 * @param zookeeperNodeParent zookeeper node path of hbase cluster. E.g, "/hbase/example-root-znode".
+	 */
+	public HBase zookeeperNodeParent(String zookeeperNodeParent) {
+		properties.putString(CONNECTOR_ZK_NODE_PARENT, zookeeperNodeParent);
+		return this;
+	}
+
+	/**
+	 * Set threshold when to flush buffered request based on the memory byte size of rows currently added . Default to <code>2mb</code>. Optional.
+	 *
+	 * @param writeBufferFlushMaxSize threshold (Byte size) to flush a buffered request. E.g, 2097152 (2MB).
+	 */
+	public HBase writeBufferFlushMaxSize(long writeBufferFlushMaxSize) {
+		properties.putLong(CONNECTOR_WRITE_BUFFER_FLUSH_MAX_SIZE, writeBufferFlushMaxSize);
+		return this;
+	}
+
+	/**
+	 * Set threshold when to flush buffered request based on the number of rows currently added.
+	 * Defaults to not set, i.e. won't flush based on the number of buffered rows. Optional.
+	 *
+	 * @param writeBufferFlushMaxRows number of added rows when begin the request flushing.
+	 */
+	public HBase writeBufferFlushMaxRows(long writeBufferFlushMaxRows) {
+		properties.putLong(CONNECTOR_WRITE_BUFFER_FLUSH_MAX_ROWS, writeBufferFlushMaxRows);
+		return this;
+	}
+
+	/**
+	 * Set a flush interval flushing buffered requesting if the interval passes, in milliseconds.
+	 * Defaults to not set, i.e. won't flush based on flush interval. Optionla.
+	 *
+	 * @param writeBufferFlushInterval flush interval in milliseconds.
+	 */
+	public HBase writeBufferFlushInterval(long writeBufferFlushInterval) {
```

Comment:
The same as above, this can be a String parameter.

Reply:
ditto.

Reply:
The Elasticsearch#bulkFlushInterval seems also be a long type..I think we also can keep the long type here for HBase.

Reply:
`Elasticsearch#bulkFlushInterval` accepts a long parameter, because the underlying `connector.bulk-flush.interval` options only accepts a long value. This is a historic problem, duration parsing in introduced recently. So I think it would be better to accept a String parameter to keep align with HBase DDL.

Reply:
The reason that `Elasticsearch#bulkFlushInterval` use long is that the properties type is just String and use `Long.valueOf`, but HBase FlushInterval is Duration. So I am +1 for String parameter.

Reply:
Got it now, will do.